syntax = "proto3";

package com.example.grpc;

message depositRequest {
    int32 customer_id = 1;

    int32 cash_amount = 2;
}

message withdrawRequest {
    int32 customer_id = 1;

    int32 cash_amount = 2;
}

message sendRequest {
    int32 customer_id = 1;

    int32 cash_amount = 2;
    
    int32 taker_id = 3;
}

message validationResponce {
    string valid = 1;
}

service bank {
    rpc deposit (depositRequest) returns (validationResponce);
    rpc withdraw (withdrawRequest) returns (validationResponce);
    rpc send (sendRequest) returns (validationResponce);
}